pkg/sanity/node.go

```diff
@@ -421,6 +434,55 @@ var _ = DescribeSanity("Node Service", func(sc *TestContext) {
 			Expect(ok).To(BeTrue())
 			Expect(serverError.Code()).To(Equal(codes.InvalidArgument), "unexpected error: %s", serverError.Message())
 		})
+
+		Describe("with single node multi writer capability", func() {
+			BeforeEach(func() {
+				if !singleNodeMultiWriterSupported {
+					Skip("Service does not have single node multi writer capability")
+				}
+			})
+
+			It("should fail when volume with single node single writer access mode is already mounted at a different target path", func() {
+				By("creating a single node single writer volume")
+				name := UniqueString("sanity-node-publish-single-node-single-writer")
+				cap := TestVolumeCapabilityWithAccessType(sc, csi.VolumeCapability_AccessMode_SINGLE_NODE_SINGLE_WRITER)
+				vol := createVolumeWithCapability(name, cap)
+
+				By("Getting a node id")
+				nid, err := r.NodeGetInfo(
+					context.Background(),
+					&csi.NodeGetInfoRequest{})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(nid).NotTo(BeNil())
+				Expect(nid.GetNodeId()).NotTo(BeEmpty())
+
+				By("Staging and publishing a volume")
+				conpubvol := controllerPublishVolumeWithCapability(name, vol, nid, cap)
+				_ = nodeStageVolumeWithCapability(name, vol, conpubvol, cap)
+				_ = nodePublishVolumeWithCapability(name, vol, conpubvol, cap)
+
+				var stagingPath string
+				if nodeStageSupported {
+					stagingPath = sc.StagingPath
+				}
+				_, err = r.NodePublishVolume(
+					context.Background(),
+					&csi.NodePublishVolumeRequest{
+						VolumeId:          vol.GetVolume().GetVolumeId(),
```

Comment:
`PublishContext` should be set if needed, as in `nodePublishVolumeWithCapability`: 
https://github.com/kubernetes-csi/csi-test/blob/9968d089294dd53c583bf11f38dafaad20ae6a93/pkg/sanity/node.go#L272-L274